src/Service/Service.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace Basis\Nats\Service;
+
+use Basis\Nats\Client;
+use Basis\Nats\Message\Payload;
+
+class Service
+{
+    public Client $client;
+
+    private string $id;
+
+    private string $name;
+
+    private string $description = '';
+
+    private string $version;
+
+    private string $started;
+
+    private array $endpoints = [];
+
+    private array $groups = [];
+
+    private array $subscriptions = [];
+
+    public function __construct(
+        Client $client,
+        string $name,
+        string $description = 'Default Description',
+        string $version = '0.0.1'
+    ) {
+        $this->client = $client;
+        $this->id = $this->generateId();
+        $this->name = $name;
+        $this->description = $description;
+        $this->version = $version;
+        $this->started = date("Y-m-d\TH:i:s.v\Z");
+
+        // Register the service verbs to listen for
+        $this->registerVerbs();
+    }
+
+    private function generateId(): string {
+        $characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+
+        $charactersLength = strlen($characters);
+
+        $randomString = "";
+
+        for ($i = 0; $i < 22; $i++) {
+            $randomString .= $characters[rand(0, $charactersLength - 1)];
+        }
+
+        return $randomString;
+    }
+
+    private function ping(): array
+    {
+        return [
+            'type' => 'io.nats.micro.v1.ping_response',
+            'name' => $this->name,
+            'id' => $this->id,
+            'version' => $this->version,
+        ];
+    }
+
+    private function info(): array
+    {
+        return [
+            'type' => 'io.nats.micro.v1.info_response',
+            'name' => $this->name,
+            'id' => $this->id,
+            'version' => $this->version,
+            'description' => $this->description,
+            'endpoints' => array_reduce($this->endpoints, function ($carry, $endpoint) {
+                $carry[] = [
+                    'name' => $endpoint->getName(),
+                    'subject' => $endpoint->getSubject(),
+                    'queue_group' => $endpoint->getQueueGroup(),
+                ];
+
+                return $carry;
+            }, []),
+        ];
+    }
+
+    private function stats(): array
+    {
+        return [
+            'type' => 'io.nats.micro.v1.stats_response',
+            'name' => $this->name,
+            'id' => $this->id,
+            'version' => $this->version,
+            'endpoints' => array_reduce($this->endpoints, function ($carry, ServiceEndpoint $endpoint) {
```

Comment:
maybe simplify this line with introduce endpoints var and fill it using foreach?